contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -88,6 +90,41 @@ contract VaultHub is PausableUntilWithRoles {
         int128 inOutDelta;
     }
 
+    /**
+     *  Obligations of the vaults towards the Lido protocol.
+     *  While any part of those obligations remains unsettled, VaultHub may want to limit what the vault can do.
+     *
+     *  Obligations have two types:
+     *  1. Redemptions.
+     *     Under extreem conditions Lido may want to rebalance the part of the vault's liability to serve the Lido Core
+     *     withdrawal queue requests to guarantee that every stETH is redeemable
+     *  2. Lido fees. Sum of infra, liquidity and reservation fees on the vault. Calculated in ether and grows on every
+     *     oracle report until paid.
+     *
+     *  Constraints until obligations settled:
```

Comment:
separate notice:
- Redemptions can't exceed liability shares (e.g. when liability shares decreased, redemptions too)

Reply:
it's even more nuanced, though